discovery-provider/src/queries/get_save_tracks.py

```diff
@@ -41,100 +45,205 @@ class GetSaveTracksArgs(TypedDict):
+        LibraryFilterType.repost: reposts_base,
+        LibraryFilterType.purchase: purchase_base,
+    }[filter_type]
+
+    # Depending on whether this is 'all' or not,
+    # subsequent query building either needs to reference TrackWithAggregates or the aliased TrackWithAggregatesAlias
+    # to avoid accidental join.
+    TracksTable = (
+        TrackWithAggregatesAlias
+        if filter_type == LibraryFilterType.all
+        else TrackWithAggregates
+    )
+
+    # Allow filtering of deletes
+    if filter_deleted:
+        base_query = base_query.filter(TracksTable.is_delete == False)
+
+    if query:
+        base_query = base_query.join(TracksTable.user, aliased=True).filter(
+            or_(
+                TracksTable.title.ilike(f"%{query.lower()}%"),
+                User.name.ilike(f"%{query.lower()}%"),
             )
         )
 
-        # Allow filtering of deletes
-        if filter_deleted:
-            base_query = base_query.filter(TrackWithAggregates.is_delete == False)
-
-        if query:
-            base_query = base_query.join(TrackWithAggregates.user, aliased=True).filter(
-                or_(
-                    TrackWithAggregates.title.ilike(f"%{query.lower()}%"),
-                    User.name.ilike(f"%{query.lower()}%"),
+    # Set sort methods
+    if sort_method == SortMethod.title:
+        base_query = base_query.order_by(sort_fn(TracksTable.title))
+    elif sort_method == SortMethod.artist_name:
+        base_query = base_query.join(TracksTable.user, aliased=True).order_by(
+            sort_fn(User.name)
+        )
+    elif sort_method == SortMethod.release_date:
+        base_query = base_query.order_by(
+            sort_fn(
+                coalesce(
+                    func.to_date_safe(
+                        TracksTable.release_date,
+                        "Dy Mon DD YYYY HH24:MI:SS",
+                    ),
+                    TracksTable.created_at,
                 )
             )
-
-        if sort_method == SortMethod.title:
-            base_query = base_query.order_by(sort_fn(TrackWithAggregates.title))
-        elif sort_method == SortMethod.artist_name:
-            base_query = base_query.join(
-                TrackWithAggregates.user, aliased=True
-            ).order_by(sort_fn(User.name))
-        elif sort_method == SortMethod.release_date:
+        )
+    elif sort_method == SortMethod.plays:
+        base_query = base_query.join(TracksTable.aggregate_play).order_by(
+            sort_fn(AggregatePlay.count)
+        )
+    elif sort_method == SortMethod.reposts:
+        base_query = base_query.join(TracksTable.aggregate_track).order_by(
+            sort_fn(AggregateTrack.repost_count)
+        )
+    elif sort_method == SortMethod.saves:
+        base_query = base_query.join(TracksTable.aggregate_track).order_by(
+            sort_fn(AggregateTrack.save_count)
+        )
+    else:
+        # This branch covers added_date, or any other sort method
```

Comment:
Combines logic here. In previous version, added_date and no method were the same, except added_date would use `sort_fn` (defaulting to `asc` if none existed) and no method always uses `desc`, even if a sort order is provided